src/BSON/Binary.c

```diff
@@ -199,6 +205,94 @@ PHP_METHOD(Binary, getType)
 }
 /* }}} */
 
+/* {{{ proto string Binary::serialize()
+*/
+PHP_METHOD(Binary, serialize)
+{
+	php_phongo_binary_t      *intern;
+#if PHP_VERSION_ID >= 70000
+	zval                      retval;
+#else
+	zval                     *retval;
+#endif
+	php_serialize_data_t      var_hash;
+	smart_str                 buf = { 0 };
+
+	intern = Z_BINARY_OBJ_P(getThis());
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+#if PHP_VERSION_ID >= 70000
+	array_init_size(&retval, 2);
+	ADD_ASSOC_STRINGL(&retval, "data", intern->data, intern->data_len);
+	ADD_ASSOC_LONG_EX(&retval, "type", intern->type);
+#else
+	ALLOC_INIT_ZVAL(retval);
+	array_init_size(retval, 2);
+	ADD_ASSOC_STRINGL(retval, "data", intern->data, intern->data_len);
+	ADD_ASSOC_LONG_EX(retval, "type", intern->type);
+#endif
+
+	PHP_VAR_SERIALIZE_INIT(var_hash);
+	php_var_serialize(&buf, &retval, &var_hash);
+	smart_str_0(&buf);
```

Comment:
Is the null byte really necessary if you're returning the string by length? Since this is primarily a function for userland, where strings are not null-terminated (i.e. null bytes aren't exposed to the users), I expect you would normally just return whatever the inner call to `serialize()` returned.